x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/response_actions/results/index.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { inspectStringifyObject } from '../../../../../utils/build_query';
+import type {
+  ActionResponsesRequestOptions,
+  ActionResponsesRequestStrategyResponse,
+  ResponseActionsQueries,
+} from '../../../../../../common/search_strategy/security_solution/response_actions';
+
+import { buildActionResultsQuery } from './query.action_results.dsl';
+import type { SecuritySolutionFactory } from '../../types';
+
+export const actionResults: SecuritySolutionFactory<ResponseActionsQueries.results> = {
+  buildDsl: (options: ActionResponsesRequestOptions) => {
+    return buildActionResultsQuery(options);
+  },
+  parse: async (options, response): Promise<ActionResponsesRequestStrategyResponse> => {
+    const inspect = {
+      dsl: [inspectStringifyObject(buildActionResultsQuery(options))],
+    };
+
+    // @ts-expect-error FIX THE TYPE
```

Comment:
Yes, but I believe this is a complex issue with overwritting or specifying a different build interface for factory. Since there were already some interfaces for security_solution.